documentation/modules/metrics/con-metrics-cruise-control.adoc

```diff
@@ -0,0 +1,96 @@
+// Module included in the following assemblies:
+//
+// metrics/assembly-metrics.adoc
+
+[id='con-metrics-cruise-control-{context}']
+= Monitoring Cruise Control operations
+
+[role="_abstract"]
+Kafka metrics are gathered by Cruise Control in order to track the utilization of brokers, topics, and partitions.
+Cruise Control also provides a set of metrics for monitoring its own performance.
+
+The Cruise Control metrics reporter collects raw metrics data from Kafka brokers.
+The data is produced to topics that are automatically created by Cruise Control.
+The metrics are used to link:{BookURLUsing}#proc-generating-optimization-proposals-str[generate optimization proposals for Kafka clusters^].
+
+Cruise Control metrics are available for real-time monitoring of Cruise Control operations.
+For example, you can use Cruise Control metrics to monitor the status of rebalancing operations that are running or provide alerts on any anomalies that are detected in an operation's performance.
+
+You can expose Cruise Control metrics by enabling JMX for Cruise Control.
```

Comment:
Thanks Jakub, I misunderstood this but you are right, we don't expose JMX since we never set KAFKA_JMX_OPTS [1] for the Cruise Control container. We only enable the Prometheus JMX Exporter [2] which "exposes Cruise Control metrics through an HTTP endpoint"

[1] https://github.com/linkedin/cruise-control/blob/migrate_to_kafka_2_4/kafka-cruise-control-start.sh#L65
[2] https://github.com/strimzi/strimzi-kafka-operator/blob/main/docker-images/kafka/cruise-control-scripts/cruise_control_run.sh#L28

Reply:
I've expanded on this a bit more, as follows.
Links to the procedure _Deploying Prometheus metrics configuration_

> You expose Cruise Control metrics by enabling the {JMXExporter} in the Cruise Control configuration.
> You can xref:proc-metrics-kafka-deploy-options-str[deploy additional metrics configuration in a ConfigMap].
> ```
> .Enabling the Prometheus JMX Exporter for Cruise Control
> [source,yaml,subs="+quotes,attributes"]
> ----
> apiVersion: {KafkaApiVersion}
> kind: Kafka
> metadata:
>   name: my-cluster
> spec:
>   # ...
>   cruiseControl:
>     metricsConfig:
>       type: jmxPrometheusExporter
>       valueFrom:
>         configMapKeyRef:
>           name: cruise-control-metrics
>           key: metrics-config.yml
> ----
> 
> ```